generator/overrides/gen.go

```diff
@@ -0,0 +1,435 @@
+						TypeInfo:         typeToOverride,
+					}
+				}
+
+				var fieldTypeToProcess *typeToProcess
+
+				switch fieldType := astField.Type.(type) {
+				case *ast.ArrayType:
+					switch elementType := fieldType.Elt.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+						if fieldTypeToProcess != nil {
+							fieldTypeToProcess.MandatoryKey = strings.Title(genutils.GetPatchMergeKey(&field))
+						}
+					}
+				case *ast.Ident:
+					fieldTypeToProcess = processFieldType(fieldType)
+					if field.Markers.Get(genutils.UnionDiscriminatorMarker.Name) != nil {
+						enumValues := []string{}
+						for _, f := range typeToOverride.Fields {
+							pos := f.RawField.Pos()
+							fieldChange := fieldChanges[f.RawField.Pos()]
+							if pos != cursor.Node().Pos() &&
+								!shouldSkip(fieldChange.OverrideMarker) {
+								enumValues = append(enumValues, fieldChange.FieldInfo.Name)
+							}
+						}
+						kubebuilderAnnotation := "+kubebuilder:validation:Enum=" + strings.Join(enumValues, ";")
+						astField.Doc = updateComments(astField, astField.Doc, `.*`, ` *`+regexp.QuoteMeta("+kubebuilder:validation:Enum=")+`.*`, kubebuilderAnnotation)
+						fieldTypeToProcess.DropEnumAnnotation = true
+					}
+				case *ast.StarExpr:
+					switch elementType := fieldType.X.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+				case *ast.MapType:
+					switch elementType := fieldType.Key.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+					switch elementType := fieldType.Value.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+					}
+				default:
+				}
+
+				if fieldTypeToProcess != nil {
+					moreTypesToAdd = append(moreTypesToAdd, *fieldTypeToProcess)
+				}
+			}
+			return true
+		},
+		func(*astutil.Cursor) bool { return true },
+	).(*ast.GenDecl)
+
+	if len(errors) > 0 {
+		return nil, moreTypesToAdd, errors // Check there is only 1 typespec in it or trigger an error
+	}
+
+	return overrideGenDecl, moreTypesToAdd, errors
+}
+
+// writeFormatted outputs the given code, after gofmt-ing it.  If we couldn't gofmt,
+// we write the unformatted code for debugging purposes.
+func (g Generator) writeOut(ctx *genall.GenerationContext, root *loader.Package, outBytes []byte) {
+	fileToWrite := "zz_generated.parent_overrides.go"
+	if g.IsForPluginOverrides {
+		fileToWrite = "zz_generated.plugin_overrides.go"
+	}
+	outputFile, err := ctx.Open(root, fileToWrite)
+	if err != nil {
+		root.AddError(err)
+		return
+	}
+	defer outputFile.Close()
+	n, err := outputFile.Write(outBytes)
+	if err != nil {
+		root.AddError(err)
+		return
+	}
+	if n < len(outBytes) {
+		root.AddError(io.ErrShortWrite)
+	}
+}
+
+func updateComments(commentedNode ast.Node, commentGroup *ast.CommentGroup, keepRegexp string, skipRegexp string, additionalLines ...string) *ast.CommentGroup {
```

Comment:
I think (but could be wrong) that this function *prepends* `additionalLines` to the comment, rather than appends. If so this should be mentioned in a doc.

Reply:
Fixed in commit https://github.com/devfile/api/pull/133/commits/37c1e341bb7bb541566dbf0b63a536b64e1b6028